app/controllers/api/v0/events_controller.rb

```diff
@@ -16,12 +19,18 @@ def create
     render nothing: true, status: 201
   end
 
-  class KafkaData < Hash
+  class KafkaData < HashWithIndifferentAccess
+    attr_reader :schema_name
+
     def initialize(api_data:, controller:)
-      super
-      merge!(api_data)
+      super()
+
+      api_data = api_data.to_hash
 
       @controller = controller
+      @schema_name = api_data.delete(:type)
```

Comment:
Removing `:type` key from the hash because with validation now turned on, AvroTurf complains if there are extra keys.